homeassistant/components/camera/generic.py

```diff
@@ -62,6 +64,7 @@ def __init__(self, hass, device_info):
         self._still_image_url = device_info[CONF_STILL_IMAGE_URL]
         self._still_image_url.hass = hass
         self._limit_refetch = device_info[CONF_LIMIT_REFETCH_TO_URL_CHANGE]
+        self._frame_interval = 1 / device_info[CONF_FRAMERATE]
```

Comment:
To prevent redundancy, I think that the framerate should be set, instead of the frame interval, and the `asyncio.sleep(self.frame_interval)` uses `asyncio.sleep(1/self.frame_rate)` instead.

Reply:
I though about doing that, but I didn't like doing the division operation every frame if I could do it only once. I don't think that it is an issue to have _frame_interval internally.

Reply:
Eh, whatever. I personally still think that is not good practice to load a value and perform the same calculation to it multiple times in different places instead of just setting it as a variable.

Reply:
Eh, whatever. I personally still think that is not good practice to load a value and perform the same calculation to it multiple times in different places instead of just setting it as a variable.

Reply:
I only do the calculation once :P

Reply:
Unless somebody adds this feature to a new platform.

Reply:
If that happens, we can have a `set_framerate(fr)` method on the base Camera platform class, that sets `self._frame_interval = 1/fr`.
Until then, I don't see an issue with having it directly on the Generic camera component.

Reply:
I guess it is alright for now